test/compress/issue-203.js

```diff
@@ -26,9 +26,7 @@ compress_new_function_with_destruct: {
     }
     expect: {
         Function("a", "[b]", "return a");
-        Function("a", "{bb}", "return a");
-        Function("[[a]]", "[{bb}]", 'return a');
+        Function("a", "{bb:b}", "return a");
+        Function("[[a]]", "[{bb:b}]", 'return a');
```

Comment:
> Do you happen to know what's the rationale for it working only if  side_effects  is disabled?

It's down to this:
```js
$ cat test.js
(function(aa, {bb}) {
    return aa;
})();

$ uglifyjs test.js -c side_effects=0
!function(aa,{bb:bb}){}();

$ uglifyjs test.js -c side_effects=1
<Nothing here but chickens!>
```
which then triggers `if (!fun) return self;` as `TreeWalker` fails to find an instance of `AST_Lambda`.

Probably means there's something to improve even on `master`, as this behaviour seems rather undesirable?